### DISCUSSION TITLE
Is there a framework-agnostic web component available for rendering BCIF?

### DISCUSSION THREAD

**Discussion by markfortner** [molstar/molstar#1217] [Q&A]
We were recently surprised when the existing mmCIF component stopped working, and we're now looking for a way of replacing that functionality. 

The existing examples seem to require React. I know there have been some comments about using Vue and giving it a div to render into, but that still seems to require React.

Is there a W3C-compliant web component available that will render the BCIF that the PDB is now serving? This type of component would be usable in React, View, Svelte, Next.js and plain HTML projects.

**Comment by dsehnal** [comment]
Hi,

> We were recently surprised when the existing mmCIF component stopped working, and we're now looking for a way of replacing that functionality.

Sorry I don't know what you are referring to.

> Is there a W3C-compliant web component available that will render the BCIF that the PDB is now serving? This type of component would be usable in React, View, Svelte, Next.js and plain HTML projects.

- Mol* offers default react UI, but depending on the complexity of you app, it is possible to use the canvas directly and write a custom UI. For example, I've previously written a completely custom UI around Mol* for [Envision](https://www.iambic-envision.com/) that doesn't use any of the built-in UI components. So the component can be written, it just requires some work.
- You can wrap the Mol* UI in a different framework, for example the [PDBe Mol* instance](https://github.com/molstar/pdbe-molstar) uses Angular.
- As for making something like this a default part of the Mol* distribution, we can definitely consider this, or at least implement it partially as a starting point for others.

**Comment by markfortner** [comment]
The URL that we're using to fetch mmCIF files from the PDB was this: https://mmtf.rcsb.org/v1.0/full/2N90 and it's now giving us an error. They may have switched URLs, we'll check into that. But the main question still remains, is there a W3C web component?

The problem with the React component is that you have to use React. If you are using any other framework, or plain HTML you're out of luck. With a standard web component, React, Vue, and plain HTML projects can all use the component. The only dependencies that are needed are the ones required for rendering the image.
